src/NServiceBus.Core/Transports/Learning/LearningTransportMessagePump.cs

```diff
@@ -186,7 +184,10 @@ await transaction.Commit()
 
                 var transportTransaction = new TransportTransaction();
 
-                transportTransaction.Set(transaction);
+                if (transactionMode == TransportTransactionMode.SendsAtomicWithReceive)
```

Comment:
This seemed like an easier way to prevent the dispatches from participating in the transaction, but maybe I'm missing some large goal here?